var/spack/repos/builtin/packages/arrayfire/package.py

```diff
@@ -64,10 +61,34 @@ def cmake_args(self):
             self.define_from_variant('AF_BUILD_CUDA', 'cuda'),
             self.define_from_variant('AF_BUILD_FORGE', 'forge'),
             self.define_from_variant('AF_BUILD_OPENCL', 'opencl'),
+            self.define_from_variant('BUILD_TESTING', 'tests'),
         ])
+
+        if '+cuda' in self.spec:
+            cuda_arch_list = self.spec.variants['cuda_arch'].value
+            cuda_arch = cuda_arch_list[0]
+            if cuda_arch == 'none':
+                args.append(self.define('CUDA_architecture_build_targets', 'Auto'))
```

Comment:
I am not sure I understand. If we don't build PTX code into the binaries, that would mean that the binary will be non functional? Do you mean that we should let the compiler select the compute architecture?

The default behavior in ArrayFire is 'Auto' so I can remove that flag. This means that if the autodetect fails(in the case the project is built in a docker), every architecture will be selected. I don't think that is desirable because it significantly increases build times. I can modify ArrayFire to let the compiler select the compute architecture if that is what spack expects.

Reply:
in HPC environments, the software is usually built at a different machine that the one with the GPU hardware. So, 'auto' detect option wouldn't help, but complicate things.

Reply:
Yes, this is my concern as well. `Auto` is the default behavior in ArrayFire. I assume packages are expected to fall back to compiler defaults if no `cuda_arch` is selected but this is not the default in ArrayFire. One approach would be to display a warning in case `none` is selected. Another would be set it to compiler defaults for `none`. I would love to hear your suggestions on the matter.

Reply:
To me, it makes sense to raise a conflict if user specifies `cuda` but not the arch like the following example from tensorflow.
```
conflicts('cuda_arch=none', when='+cuda', msg='Must specify CUDA compute capabilities of your GPU, see https://developer.nvidia.com/cuda-gpus')
```